### PR TITLE
Refactored the GMFs

### PR DESCRIPTION
**Pull request by micheles**
Reflecting the changes in hazardlib https://github.com/gem/oq-hazardlib/pull/450. Also, the way the GMFs are stored is different, since now there is an `imti` index. This is a first step towards https://github.com/gem/oq-engine/issues/2017. The big difference with respect to master is that now the event based hazard calculator discards the value below the mininum intensity (before only the risk calculator had this feature).
Here is how big the effect is for a reduced computation for Turkey:

```
   master (no filtering)
   operation                time_sec memory_mb counts
   total compute_gmfs_and_curves 5,068    668       27    
   compute poes                  4,998    280       60,953
   saving gmfs                   10       2.637     1,728 
   gmf_data                                         6.03 GB  

   this branch (with filtering)
   operation                time_sec memory_mb counts
   total compute_gmfs_and_curves  2,391     6.332     27     
   compute poes                   2,262     0.332     60,953 
   saving gmfs                    2.026     10        1,705  
   gmf_data                                         80.41 MB 
```

Thanks to the refactoring in the hazardlib the performance of the risk calculator when computing the GMFs has improved too (nearly half memory and twice the speed):

```
master:   3,984 s, 121 MB
this branch:  2,141 s, 76 MB
```
The tests are running here: https://ci.openquake.org/job/zdevel_oq-risklib/1560

### CODE REVIEW

File: openquake/commonlib/export/hazard.py [hunk 1]
```diff
@@ -604,15 +599,15 @@ def _get_gmfs(dstore, serial, eid):
...
+                      oq.truncation_level, correl_model, min_iml).by_rlzi()
+    for imti, imt in enumerate(oq.imtls):
         gmfa = numpy.zeros(N, gmf_dt)
-        for rlz in rlzs:
-            data = gmf_list[rlz.ordinal]
-            ok = data[data['eid'] == eid]
-            gmfa['%03d' % rlz.ordinal][rup.indices] = ok['gmv'][imt]
+        for rlzname in gmf_dt.names:
+            rlzi = int(rlzname)
+            gmvs = get_array(gmfadict[rlzi], imti=imti)['gmv']
```
Comment (ptormene) [inline openquake/commonlib/export/hazard.py]: Please check if we need to filter also by eid

File: openquake/risklib/riskinput.py [hunk 2]
```diff
@@ -16,26 +16,22 @@
...
 from openquake.baselib.general import groupby, split_in_blocks
-from openquake.hazardlib.calc.gmf import GmfComputer
-from openquake.hazardlib import site
+from openquake.hazardlib import site, calc
 from openquake.risklib import scientific, riskmodels
 
 U32 = numpy.uint32
 F32 = numpy.float32
 
+gmv_eid_dt = numpy.dtype([('gmv', F32), ('eid', U32)])
```
Comment (ptormene) [inline openquake/risklib/riskinput.py]: Please check if this is used

File: openquake/commonlib/export/hazard.py (hunk 1 above)
Comment (micheles) [inline openquake/commonlib/export/hazard.py]: We need. Added.

File: openquake/risklib/riskinput.py (hunk 2 above)
Comment (micheles) [inline openquake/risklib/riskinput.py]: It is not needed

**Comment by ptormene** [comment]
LGTM

**Pull request merged**